pandas/tests/window/test_rolling.py

```diff
@@ -1500,3 +1500,31 @@ def test_rolling_numeric_dtypes():
         dtype="float64",
     )
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("window", [1, 3, 10, 1000])
+def test_rank(window):
+    length = 1000
+    ser = Series(data=np.random.rand(length))
+
+    result = Series([
+        ser[i - window:i].rank().iloc[-1] if i - window >= 0 else np.NaN
+        for i in range(1, length + 1)
```

Comment:
can you compare vs .rolling.appyl(lambda x: x.rank()) vs this?

Reply:
done